### ISSUE TITLE
Media: Clicking Post/Update before upload complete causes infinite wait

### ISSUE THREAD

**Issue by hoverduck** [wordpress-mobile/WordPress-Editor-iOS#795]
When adding an image to a post, if I click the Post/Update button before it's done I get the alert saying to please wait until it's complete.  But then even when the progress bar across the top of the image indicates that it's done, the top left of the screen says it's still "Uploading...", and never changes.

https://cloudup.com/cLvbtoAZfP7

**Comment by designsimply** [comment]
Tested and confirmed: [1m25s](https://cloudup.com/cqHxqKhHSro)

Note: the problem happens for me no matter what icon I click in the top bar, not just Post/Update.

**Comment by SergioEstevao** [comment]
Did you test this on the latest 6.1 internal release?

**Comment by designsimply** [comment]
@SergioEstevao, I didn't test with the 6.1 internal release earlier—tested it just now, and it's broken in a different way. I am able to interact with the app if I tap the top icons in a post while images are uploading and I can publish the post, but the images end up broken in the post preview and live post.

Video: [3m12s](https://cloudup.com/cA2C76viUWI) (tested with iOS 9.3 iPhone 6S WPiOS6.1.0.20160318)

Posts for ref:

* http://watchingstormsrollby.me/2016/03/28/more-images/
* http://watchingstormsrollby.me/2016/03/28/one-more-try/

![screen shot 2016-03-28 at mon mar 28 4 28 28 pm](https://cloud.githubusercontent.com/assets/1119271/14092152/35f65220-f502-11e5-96b4-2e1c57b98edf.png)
<sup>Seen at http://watchingstormsrollby.me/2016/03/28/more-images/ using Chrome 49.0.2623.87 on Mac OS X 10.11.13</sup>

Screenshot showing a broken image from the "more images" test post:

`src` looks wonky:

```
<span id="img_container_F40FD4FC-77D6-405A-8A42-1E4BC5E7A10F" class="img_container"><img src="/var/mobile/Containers/Data/Application/897A7E74-9AE1-42D6-97A6-DB8D86DC0BBB/Documents/Media/img_3647-thumbnail.jpeg" alt="" class="uploading" scale="0"></span>
```

**Comment by SergioEstevao** [comment]
Ok, I think we have that fixed for the next release of the 6.1 build, I will keep an eye on it.
